client/go/outline/electron/go_plugin.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+/*
+#include <stdlib.h>  // for C.free
+
+// InvokeGoAPIResult is a struct used to pass result from Go to TypeScript boundary.
+typedef struct InvokeGoAPIResult_t
```

Comment:
I think you can define this struct in Go instead and only have the include here.

Reply:
Do you mean define it as the `struct` as:

```go
type InvokeGoAPIResult struct {
  Output, ErrorJson *C.char
}
```

I'm afraid this is [not supported by CGo](https://pkg.go.dev/cmd/cgo#hdr-C_references_to_Go):

> Not all Go types can be mapped to C types in a useful way. Go struct types are not supported; use a C struct type. Go array types are not supported; use a C pointer.